toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt

```diff
@@ -126,42 +132,97 @@ public class TraceState(
         _selectedTraceConfiguration.value = config
     }
 
+    private var _currentTraceRun: TraceRun? = null
+    internal val currentTraceRun: TraceRun
+        get() = _currentTraceRun ?: throw IllegalStateException("TraceRun cannot be null")
+
+    private val currentTraceGraphics : MutableList<Graphic> = mutableListOf()
+
     /**
-     * TBD
+     * Run a trace on the Utility Network using the selected trace configuration and starting points.
+     *
+     * @return true if the trace results are available, false otherwise.
+     * @since 200.6.0
      */
-    public suspend fun trace() {
+    public suspend fun trace() : Boolean {
```

Comment:
this can be internal
```suggestion
    internal suspend fun trace() : Boolean {
```

Reply:
This function should only throw CancellationExceptions or it could bring down the whole app. This can be fixed by surrounding the calling site in a try-catch, making this function a try-catch expression, or making call `runCatchingCancellable`.

Reply:
going with - surrounding the calling site in a try-catch